x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -130,6 +130,7 @@ RP : ')';
 TRUE : 'true';
 
 EQ  : '==';
+EQ_IGNORE_CASE  : '=~';
```

Comment:
I propose we use SEQ for "string equality" (just like in EQL)  or IEQ for "Insensitive Equality".
Same for the backing class.